managed-ledger/src/main/java/org/apache/bookkeeper/mledger/ManagedCursor.java

```diff
@@ -150,9 +150,11 @@ enum IndividualDeletedEntries {
      *            opaque context
      * @param maxPosition
      *            max position can read
+     * @param skipCondition
+     *            predicate of read filter out
      */
     void asyncReadEntries(int numberOfEntriesToRead, ReadEntriesCallback callback, Object ctx,
-                          PositionImpl maxPosition);
+                          PositionImpl maxPosition, Predicate<PositionImpl> skipCondition);
```

Comment:
It's better to add a new method.
I mean, we'd better not change the existing API directly.
This class is annotated with 

```
@InterfaceAudience.LimitedPrivate
@InterfaceStability.Stable
```

Reply:
Good idea.